openedx/core/djangoapps/content_libraries/models.py

```diff
@@ -0,0 +1,82 @@
+"""
+Models for new Content Libraries
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from django.contrib.auth import get_user_model
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+import six
+
+User = get_user_model()
+
+
+@six.python_2_unicode_compatible  # pylint: disable=model-missing-unicode
+class ContentLibrary(models.Model):
+    """
+    A Content Library is a collection of content (XBlocks and/or static assets)
+
+    All actual content is stored in Blockstore, and any data that we'd want to
+    transfer to another instance if this library were exported and then
+    re-imported on another Open edX instance should be kept in Blockstore. This
+    model in the LMS should only be used to track settings specific to this Open
+    edX instance, like who has permission to edit this content library.
+    """
+    # Every Library is uniquely and permanently identified by a slug that is set
+    # during creation/import. The slug will appear in the library's opaque key:
+    # e.g. "lib:slug" is the opaque key for a library wtih slug "slug"
+    slug = models.SlugField(primary_key=True)
+    bundle_uuid = models.UUIDField(unique=True, null=False)
+
+    # TODO: Who is authorized to work on this library (in addition to permission inherited from its collection)
+    # -> Define a CollectionPermissions model with m2m field to AuthorizedUser(user_id, role: Admin/Author/Collaborator)
+
+    # How is this library going to be used?
+    allow_public_learning = models.BooleanField(
+        default=False,
+        help_text=("""
+            Allow any user (even unregistered users) to view and interact with
+            content in this library (in the LMS; not in Studio). If this is not
+            enabled, then the content in this library is not directly accessible
+            in the LMS, and learners will only ever see this content if it is
+            explicitly added to a course. If in doubt, leave this unchecked.
+        """),
+    )
+    allow_public_read = models.BooleanField(
+        default=False,
+        help_text=("""
+            Allow any user with Studio access to view this library's content in
+            Studio, use it in their courses, and copy content out of this
+            library. If in doubt, leave this unchecked.
+        """),
+    )
```

Comment:
I suspect that we'll want to keep this model pretty bare and eventually the licensing/rights will be another model or set of models that reference this one. That being said, I'm fine with having these fields here as we're starting out -- I just don't want to end up with the kitchen sink that is CourseOverviews.

Are these both use cases that we need on day 1? They seem perfectly reasonable, but I'm fine with defaulting everything to private to a user or org and not even giving the option if that's not currently something we have a concrete use case for.

Reply:
We need the "Allow public learning" permission for day 1 for LabXchange, but not necessarily the other one.

And yeah, I was hoping that ownership, permissions, licensing, etc. would be managed through another "Collection" model, but at the moment I realized I don't have those product requirements nailed down and I'm still unclear if that Collection model would best live in Blockstore or in the LMS (I'm now thinking LMS makes more sense, as Blockstore data should export/import between blockstores and anything user-specific would not?)